### PR TITLE
add auto init for apx v2

### PR DESCRIPTION
**Pull request by CosmicFusion**

### CODE REVIEW

File: command/command.go [hunk 1]
```diff
@@ -55,6 +75,53 @@ func (c *Command) processCommand() error {
 	}
 
 	var err error
+	
+	if c.OsType != types.Ubuntu && c.OsType != types.Flatpak && c.Command != "init" && c.ContainerName == "" {
+		cmd := exec.Command("/bin/bash", "-c", "apx subsystems list | grep ^" + c.ContainerSubsystem() + "$")
+    		cmd.Stdin = os.Stdin
+    		cmd.Stderr = os.Stderr
+    		if err != nil {
+       			fmt.Println("Error 1")
```
Comment (ferrreo) [inline command/command.go]: err will always be nil here so this does nothing

File: command/command.go [hunk 2]
```diff
@@ -55,6 +75,53 @@ func (c *Command) processCommand() error {
...
+    		}
+    		if err := cmd.Start(); err != nil {
+        		fmt.Println("Error 2")
+    		}
+    		if err != nil {
+        		fmt.Println("Error 3")
```
Comment (ferrreo) [inline command/command.go]: already checked the error in the previous if don't need to do it again here

File: command/command.go [hunk 3]
```diff
@@ -55,6 +75,53 @@ func (c *Command) processCommand() error {
...
+	
+	if c.OsType != types.Ubuntu && c.OsType != types.Flatpak && c.Command != "init" && c.ContainerName == "" {
+		cmd := exec.Command("/bin/bash", "-c", "apx subsystems list | grep ^" + c.ContainerSubsystem() + "$")
+    		cmd.Stdin = os.Stdin
+    		cmd.Stderr = os.Stderr
+    		if err != nil {
+       			fmt.Println("Error 1")
+    		}
+    		if err := cmd.Start(); err != nil {
+        		fmt.Println("Error 2")
```
Comment (ferrreo) [inline command/command.go]: This should just be `return err`

File: command/command.go [hunk 4]
```diff
@@ -55,6 +75,53 @@ func (c *Command) processCommand() error {
...
+    		}
+    		if err != nil {
+        		fmt.Println("Error 3")
+    		}
+    		if err := cmd.Wait(); err != nil {
+    			fmt.Println("Warning: Subsystem hasn't been pre-initialized, initializing...")
+        		cmd_exec := exec.Command("/bin/bash", "-c", "apx subsystems new -n " + c.ContainerSubsystem() + " -s " + c.ApxSubsystem())
+        		cmd_exec.Stdout = os.Stdout
+			if err := cmd_exec.Run(); err != nil {
+  				fmt.Println("Apx Error: ", err)
```
Comment (ferrreo) [inline command/command.go]: need to return here

File: command/command.go [hunk 5]
```diff
@@ -55,6 +75,53 @@ func (c *Command) processCommand() error {
...
+			}
+    		}
+	}
+	
+	if c.OsType != types.Ubuntu && c.OsType != types.Flatpak && c.Command != "init" && c.ContainerName != "" {
+		cmd := exec.Command("/bin/bash", "-c", "apx subsystems list | grep ^" + c.ContainerName + "$")
+    		cmd.Stdin = os.Stdin
+    		cmd.Stderr = os.Stderr
+    		if err != nil {
+       			fmt.Println("Error 1")
```
Comment (ferrreo) [inline command/command.go]: err will always be nil here this does nothing

File: command/command.go [hunk 6]
```diff
@@ -55,6 +75,53 @@ func (c *Command) processCommand() error {
...
+    		}
+    		if err := cmd.Start(); err != nil {
+        		fmt.Println("Error 2")
```
Comment (ferrreo) [inline command/command.go]: just do `return err`

File: command/command.go [hunk 7]
```diff
@@ -55,6 +75,53 @@ func (c *Command) processCommand() error {
...
+    		}
+    		if err != nil {
+        		fmt.Println("Error 3")
```
Comment (ferrreo) [inline command/command.go]: err already checked so this is pointless

File: command/command.go [hunk 8]
```diff
@@ -55,6 +75,53 @@ func (c *Command) processCommand() error {
...
+    		}
+    		if err := cmd.Wait(); err != nil {
+    			fmt.Println("Warning: Subsystem hasn't been pre-initialized, initializing...")
+        		cmd_exec := exec.Command("/bin/bash", "-c", "apx subsystems new -n " + c.ContainerName + " -s " + c.ApxSubsystem())
+        		cmd_exec.Stdout = os.Stdout
+			if err := cmd_exec.Run(); err != nil {
+  				fmt.Println("Apx Error: ", err)
```
Comment (ferrreo) [inline command/command.go]: need to return here

File: command/command.go [hunk 9]
```diff
@@ -55,6 +75,53 @@ func (c *Command) processCommand() error {
...
+    		if err := cmd.Start(); err != nil {
+        		fmt.Println("Error 2")
+    		}
+    		if err != nil {
+        		fmt.Println("Error 3")
+    		}
+    		if err := cmd.Wait(); err != nil {
+    			fmt.Println("Warning: Subsystem hasn't been pre-initialized, initializing...")
+        		cmd_exec := exec.Command("/bin/bash", "-c", "apx subsystems new -n " + c.ContainerSubsystem() + " -s " + c.ApxSubsystem())
+        		cmd_exec.Stdout = os.Stdout
```
Comment (ferrreo) [inline command/command.go]: need to assign stdin here too so user can enter sudo details if needed

File: command/command.go [hunk 10]
```diff
@@ -55,6 +75,53 @@ func (c *Command) processCommand() error {
...
+    		if err := cmd.Start(); err != nil {
+        		fmt.Println("Error 2")
+    		}
+    		if err != nil {
+        		fmt.Println("Error 3")
+    		}
+    		if err := cmd.Wait(); err != nil {
+    			fmt.Println("Warning: Subsystem hasn't been pre-initialized, initializing...")
+        		cmd_exec := exec.Command("/bin/bash", "-c", "apx subsystems new -n " + c.ContainerName + " -s " + c.ApxSubsystem())
+        		cmd_exec.Stdout = os.Stdout
```
Comment (ferrreo) [inline command/command.go]: need to assign stdin here too so user can enter sudo details if needed

File: command/command.go [hunk 11]
```diff
@@ -55,6 +75,53 @@ func (c *Command) processCommand() error {
...
+       			fmt.Println("Error 1")
+    		}
+    		if err := cmd.Start(); err != nil {
+        		fmt.Println("Error 2")
+    		}
+    		if err != nil {
+        		fmt.Println("Error 3")
+    		}
+    		if err := cmd.Wait(); err != nil {
+    			fmt.Println("Warning: Subsystem hasn't been pre-initialized, initializing...")
```
Comment (ferrreo) [inline command/command.go]: We shouldn't be using an error to do this logic, instead parse the output of the command and react accordingly. Other actual errors could also happen e.g broken podman where we don't want to then call the subsystem new command.

File: command/command.go [hunk 12]
```diff
@@ -55,6 +75,53 @@ func (c *Command) processCommand() error {
...
```
Comment (ferrreo) [inline command/command.go]: We shouldn't be using an error to do this logic, instead parse the output of the command and react accordingly. Other actual errors could also happen e.g broken podman where we don't want to then call the subsystem new command.

File: command/command.go [hunk 13]
```diff
@@ -55,6 +75,53 @@ func (c *Command) processCommand() error {
...
+        		cmd_exec := exec.Command("/bin/bash", "-c", "apx subsystems new -n " + c.ContainerSubsystem() + " -s " + c.ApxSubsystem())
+        		cmd_exec.Stdout = os.Stdout
+			if err := cmd_exec.Run(); err != nil {
+  				fmt.Println("Apx Error: ", err)
+			}
+    		}
+	}
+	
+	if c.OsType != types.Ubuntu && c.OsType != types.Flatpak && c.Command != "init" && c.ContainerName != "" {
+		cmd := exec.Command("/bin/bash", "-c", "apx subsystems list | grep ^" + c.ContainerName + "$")
```
Comment (ferrreo) [inline command/command.go]: shouldn't be using grep, just parse it in golang

File: command/command.go [hunk 14]
```diff
@@ -55,6 +75,53 @@ func (c *Command) processCommand() error {
 	}
 
 	var err error
+	
+	if c.OsType != types.Ubuntu && c.OsType != types.Flatpak && c.Command != "init" && c.ContainerName == "" {
+		cmd := exec.Command("/bin/bash", "-c", "apx subsystems list | grep ^" + c.ContainerSubsystem() + "$")
```
Comment (ferrreo) [inline command/command.go]: shouldn't be using grep, just parse it in golang

**Review by ferrreo — CHANGES_REQUESTED** [review]

File: command/command.go (hunk 10 above)
Comment (CosmicFusion) [reply command/command.go]: apx doesn't require sudo

Comment (ferrreo) [reply command/command.go]: should still assign it as it might not always be the case

**Review by ferrreo — APPROVED** [review]

**Pull request merged**